signals/src/signal.rs

```diff
@@ -450,6 +458,109 @@ impl<A> Signal for Flatten<A>
     }
 }
 
+#[doc(hidden)]
+pub struct Broadcast<A: Signal>
+    where <A as Signal>::Item: Clone {
+
+    // TODO: should these be Arcs ??
+
+    // The state of this instance
+    state: Rc<RefCell<Option<<A as Signal>::Item>>>,
+
+    // The state of the whole broadcast
+    inner: Rc<RefCell<BroadcastInner<A>>>
+}
+
+impl<A: Signal> Broadcast<A>
+    where <A as Signal>::Item: Clone {
+    fn new(signal: A) -> Self {
+        let state = Rc::new(RefCell::new(None));
+        let inner = BroadcastInner {
+            signal: signal,
+            children: vec![Rc::downgrade(&state)]
```

Comment:
`Future::shared` uses a `HashMap`, I'm not sure whether `HashMap` or `Vec` is better.

Reply:
My instinct was that with Vec we could get away without having IDs. This came at the downside of having the two-Rc solution. Maybe the hashmap is the way.

Reply:
Using `Vec` means that it will retain memory in the `Vec` until it is `poll`d, with a `HashMap` it can remove the memory immediately when the `Broadcast` is dropped.

It's a minor detail, but I like being thorough.

Reply:
I do really like the simplicity of `Vec` though, so I'm a bit torn.